mbodied/agents/backends/gradio_backend.py

```diff
@@ -51,9 +53,10 @@ def async_act(self, *args, api_name="/predict", result_callbacks=None, blocking=
             Job: Gradio job object.
         """
         job = self.client.submit(api_name=api_name, result_callbacks=result_callbacks, *args, **kwargs)
-        tic = time.time()
         if blocking:
+            tic = time.time()
             while not job.done:
```

Comment:
just call job.result(timeout=timeout) if its blocking.

Reply:
This method should not have a blocking at all. Updated and also renamed.